src/main/java/com/uipath/uipathpackage/UiPathPack.java

```diff
@@ -59,25 +63,36 @@ public UiPathPack(Entry version, String projectJsonPath, String outputPath) {
-        EnvVars env = run.getEnvironment(listener);
+        EnvVars envVars = run.getEnvironment(listener);
         Utility util = new Utility();
+        FilePath tempRemoteDir = tempDir(workspace);
+        tempRemoteDir.mkdirs();
+        FilePath tempOutputDir = tempRemoteDir.child("Output");
+        tempOutputDir.mkdirs();
         util.validateParams(projectJsonPath, "Invalid Project Json Path");
         util.validateParams(outputPath, "Invalid Output Path");
-        String projectPathFormatted = escapePowerShellString(env.expand(projectJsonPath.trim()));
-        String outputPathFormatted = escapePowerShellString(env.expand(outputPath.trim()));
+        FilePath projectPath = new FilePath(new File(envVars.expand(projectJsonPath)));
+        FilePath outputDir = new FilePath(new File(envVars.expand(outputPath)));
+
         try {
-            String importModuleCommands = util.importModuleCommands(listener, env);
+            String importModuleCommands = util.importModuleCommands(tempRemoteDir, listener, envVars);
             String generatePackCommand;
             if (version instanceof ManualEntry) {
-                String versionFormatted = escapePowerShellString(env.expand(((ManualEntry) version).getText().trim()));
-                generatePackCommand = String.format("Pack -projectJsonPath %s -packageVersion %s -outputFolder %s", projectPathFormatted, versionFormatted, outputPathFormatted);
+                String versionFormatted = escapePowerShellString(envVars.expand(((ManualEntry) version).getText().trim()));
+                generatePackCommand = String.format("Pack -projectJsonPath %s -packageVersion %s -outputFolder %s", escapePowerShellString(projectPath.getRemote()), versionFormatted, escapePowerShellString(tempOutputDir.getRemote()));
             } else {
-                generatePackCommand = String.format("Pack -projectJsonPath %s -outputFolder %s", projectPathFormatted, outputPathFormatted);
+                generatePackCommand = String.format("Pack -projectJsonPath %s -outputFolder %s", escapePowerShellString(projectPath.getRemote()), escapePowerShellString(tempOutputDir.getRemote()));
+            }
+            if (!new PowerShell(getCommand(importModuleCommands, generatePackCommand)).perform((AbstractBuild<?, ?>) run, launcher, listener)) {
+                throw new AbortException("Failed to execute powershell session");
             }
-            util.execute(listener, importModuleCommands, generatePackCommand);
-        } catch (IOException | URISyntaxException e) {
+            //copy result to outputDir
+            tempRemoteDir.child("Output").copyRecursiveTo(outputDir);
+        } catch (URISyntaxException e) {
             e.printStackTrace(listener.getLogger());
             throw new AbortException(e.getMessage());
+        } finally {
+            tempRemoteDir.deleteRecursive();
```

Comment:
All comments in UiPathDeploy.java related to exception handling and cleanup are applicable for this file as well.